### DISCUSSION TITLE
How to disable scip generation for Scala 3 and/or Scala.js in a cross-built project?

### DISCUSSION THREAD

**Discussion by yoohaemin** [ckipp01/mill-scip#59] [Q&A]
Hi, thanks for the great project.
I recently switched one of my projects from sbt to mill, and added sourcegraph.yml for github actions (copied over from here), and it works really well.
However I noticed that it tries to build semanticdb for the whole matrix for pure cross-built projects. Doing it for a scala.js project is unnecessary, and the Scala 3 one is apparently less featureful (https://sourcegraph.github.io/scip-java/docs/getting-started.html#scala).

Could you provide a hint to update the CI script so that it doesn't do redundant work? 
https://github.com/yoohaemin/decrel/blob/master/.github/workflows/sourcegraph.yml

Example run: https://github.com/yoohaemin/decrel/actions/runs/4459131729/jobs/7831389763?pr=52
My repo: https://github.com/yoohaemin/decrel

**Comment by ckipp01** [comment]
Hey @yoohaemin! Good question. There currently actually isn't a way to do this. I wanted to avoid users having to add anything to their build to use this and the easiest way to accomplish this was to just do

```scala
    ev.rootModule.millInternal.modules.collect { case j: JavaModule => j }
```

The negative part of this is that it just grabs everything. There is a couple things that we could do here:

1. Add an exclusion list to `generate` which would filter out any named modules
2. Add an includes list to `generate` which tells it which to include

Not fully sure which would be best without really exploring it.

**Reply by yoohaemin** [reply]
Gotcha. 
I think we do want to have this much flexibility right? Maybe I can come up with a PR that adds both exclude and include-only list (since I also don't have visibility on what's a better one).
